core/gatewayclient/client.go

```diff
@@ -189,7 +190,11 @@ func (c *GatewayClient) GetAuthCookie() (cookie *http.Cookie, found bool) {
 // If the cookie is not valid for the client target, it returns an error.
 // It will overwrite any existing authentication cookie.
 func (c *GatewayClient) SetAuthCookie(cookie *http.Cookie) error {
-	if cookie.Domain != "" && cookie.Domain != c.target.Host {
+	// ref https://stackoverflow.com/a/16328399.
+	// KGW already set the cookie domain without port in
+	// https://github.com/kwilteam/kgw/pull/18/files#diff-5b365c916e8b28d0115f136435a03daa1ef6df8cf0eb49479c556923545b56c7R60
+	targetDomain := strings.Split(c.target.Host, ":")[0]
+	if cookie.Domain != "" && cookie.Domain != targetDomain {
```

Comment:
Yeah, good catch.
Fix seems fine.  Only call SetAuthCookie once client use in most cases, so no strong need to store this `targetDomain` in a field.  Also could/should use `strings.Cut` (see docs for `strings.Split`) but just a nit.

Reply:
hmm, if `strings.Cut` then I think we'll need `targetDomain`? As it returns multi values

Or `if cookie.Domain != "" && cookie.Domain != strings.Split(c.target.Host, ":")[0]` ?

Reply:
It's good as-is